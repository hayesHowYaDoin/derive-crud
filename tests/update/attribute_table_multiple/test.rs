use derive_crud::Update;

#[derive(Update)]
#[crud_table("test_table")]
#[crud_table("test_table_2")]
struct Test {
    #[crud_id]
    id: i64,
    name: String,
}

#[tokio::main]
async fn main() {
    let _result = Test {
        id: 1,
        name: "Debbie".to_string(),
    }
    .update()
    .await;
}
