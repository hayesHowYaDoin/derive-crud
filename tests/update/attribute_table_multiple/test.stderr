error: Only one `#[crud_table]` attribute is allowed
 --> tests/update/attribute_table_multiple/test.rs:5:3
  |
5 | #[crud_table("test_table_2")]
  |   ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no method named `update` found for struct `Test` in the current scope
  --> tests/update/attribute_table_multiple/test.rs:21:6
   |
6  |   struct Test {
   |   ----------- method `update` not found for this struct
...
17 |       let _result = Test {
   |  ___________________-
18 | |         id: 1,
19 | |         name: "Debbie".to_string(),
20 | |     }
21 | |     .update(&pool)
   | |     -^^^^^^ method not found in `Test`
   | |_____|
   |
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `update`, perhaps you need to implement one of them:
           candidate #1: `digest::Update`
           candidate #2: `digest::digest::Digest`
           candidate #3: `digest::digest::DynDigest`
